python/ql/src/Imports/UnusedImport.ql

```diff
@@ -41,6 +41,37 @@ predicate all_not_understood(Module m) {
     )
 }
 
+predicate imported_module_used_in_doctest(Import imp) {
+    exists(string modname |
+        ((Name)imp.getAName().getAsname()).getId() = modname
+        and
+        /* Look for doctests containing the patterns:
+         * >>> …name…
+         * ... …name…
+         */
+        exists(StrConst doc |
+            doc.getEnclosingModule() = imp.getScope() and
+            doc.isDocString() and
+            doc.getText().regexpMatch("[\\s\\S]*(>>>|\\.\\.\\.).*" + modname + "[\\s\\S]*")
+        )
+    )
+}
+
+predicate imported_module_used_in_typehint(Import imp) {
+    exists(string modname |
+        ((Name)imp.getAName().getAsname()).getId() = modname
```

Comment:
Any reason for not using an infix cast? (Also, the `.getAName().getAsname().(Name)` invocation always makes me pause. Surely we could make this a bit less redundant?)